osi_sensorview.proto

```diff
@@ -172,6 +172,43 @@ message GenericSensorView
     optional GenericSensorViewConfiguration view_configuration = 1;
 }
 
+//
+// \brief Obstruction values for a given object and sensor.
+//
+// For a given object in the ground truth data, and for a particular sensor, how
+// obstructed is that object from the point of view of that sensor.
+//
+message Obstruction
+{
+    // ID of the object in ground truth data
+    //
+    // This field is required as without it the entire message ships
+    // no useful information.
+    //
+    optional Identifier ground_truth_id = 1;
+
+    // Total visible proportion for the object.
+    //
+    // This value considers various factors that would stop an object from being
+    // visible from the point of view of this sensor.  It should include
+    // truncation (outside of the field of view of the sensor), distance (beyond
+    // the range of the sensor), and occlusion (how hidden is the object by
+    // other objects in the scene).
+    //
+    // Occlusion can be caused by both static objects (trees, buildings, etc.)
+    // and dynamic objects (cars, pedestrians, etc.).  By convention
+    // semi-transparent surfaces (e.g. car windows) are considered opaque.
+    //
+    // \note The object is considered to be the 2D cross-section of the object
+    // as visible from the sensor.
+    //
+    // The value is in the range 0.0 (fully hidden) and 1.0 (fully visible).
+    //
+    // \image html OSI_Obstruction_visibility.svg "Obstruction visibility"
+    //
+    optional double visibility = 2;
+}
```

Comment:
I would ask for the absolute projected area in sqm of the projection onto the Z-plane (e.g. sphere or cylinder at 1 m).